reco-service/src/main/java/com/movie/recommender/reco/service/RecommendationService.java

```diff
@@ -0,0 +1,102 @@
+package com.movie.recommender.reco.service;
+
+import com.movie.recommender.common.client.FavoritesClient;
+import com.movie.recommender.common.client.LocationServiceClient;
+import com.movie.recommender.common.client.MovieRecoClient;
+import com.movie.recommender.common.model.location.LocationDTO;
+import com.movie.recommender.common.model.movie.NowPlayingMoviesByCountry;
+import com.movie.recommender.common.model.reco.PredictRequest;
+import com.movie.recommender.common.model.reco.PredictResponse;
+import com.movie.recommender.common.model.reco.Prediction;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import lombok.extern.slf4j.Slf4j;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class RecommendationService {
+
+    private final LocationServiceClient locationClient;
+    private final FavoritesClient favoritesClient;
+    private final MongoDBService mongoDBService;
+    private final MovieRecoClient movieRecoClient;
+    private final String COUNTRY_CODE = "ua";
+    @Value("${reco.threshold}")  // Read the threshold value from the configuration file
+    private double scoreThreshold;
+
+    public RecommendationService(
+            LocationServiceClient locationClient,
+            FavoritesClient favoritesClient,
+            MongoDBService mongoDBService,
+            MovieRecoClient movieRecoClient
+    ) {
+        this.locationClient = locationClient;
+        this.favoritesClient = favoritesClient;
+        this.mongoDBService = mongoDBService;
+        this.movieRecoClient = movieRecoClient;
+    }
+
+    public List<Prediction> getRecommendations(Long userId) {
+        log.info("Starting recommendation process for user ID: {}", userId);
+
+        LocationDTO location = fetchUserLocation(userId);
+        Set<Long> favoriteMovies = fetchUserFavoriteMovies(userId);
+        List<Long> nowPlayingMovieIds = fetchNowPlayingMovieIds();
+
+        PredictRequest predictRequest = createPredictRequest(favoriteMovies, nowPlayingMovieIds);
+        PredictResponse predictResponse = fetchPredictions(predictRequest);
+
+        return filterPredictions(predictResponse, scoreThreshold);
```

Comment:
after filtering predictions for the filtered movies, retrieve the corresponding movie details from the movieDB and return them